[DNM] Hydrogen Gas

DaedalusGame opened the pull request:
:cl: DaedalusGame, More Robust Than You, Buggy123
add: Added Hydrogen Gas
add: Added Water Vapor Synthesis
/:cl:

- [x] Initial Gas
- [x] Water Vapor Synthesis
- [x] Hydrogen Explosion
- [ ] Cleanup Reaction Code
- [ ] Visuals for Synthesis (Making it looks like fire requires extra changes)
- [x] Canister forme (Sprites needed)
- [ ] Supermatter Interaction (?)
- [ ] Improve Hydrogen Burn formula
- [ ] Tesla Electrolysis
- [x] Hydrogen Reagent dumping out as gas

[Hydrogen Explosion in Action](https://i.imgur.com/bYFm5VF.gifv)

DaedalusGame committed:
- Adds hydrogen gas
- Adds water vapor synthesis from hydrogen, heavily exothermic
- Adds hydrogen canister

nicbn commented on code/modules/atmospherics/gasmixtures/reactions.dm:
```diff
@@ -357,13 +361,70 @@
+
+			if(cached_results[id] >= 0)
+				if(istype(location))
+					explosion(location,0,0,2,adminlog = FALSE,flame_range = 2) //If we log this oranges will yell at me
```
The ultimate lag

DaedalusGame replied:
it works great!

nicbn replied:
Also I think explosions should scale according to partial pressure of H2+O2, creating hull breaches if possible.

Spaceinaba commented:
now we just need carbon so we can make propane and atmos techs can become hank hill

Buggy123 commented:
Plasma is already sorta carbon ish? It burns into carbon dioxide, if nothing else.

DaedalusGame commented:
According to the defines, tritium is the most carbon you can get

Tlaltecuhtli commented:
how do u get water? heating water to 100K?

Tlaltecuhtli commented:
water vapour*

DaedalusGame commented:
You burn hydrogen with oxygen into water vapor

arsserpentarium commented:
water should be able to be separated to oxygen and hydrogen

DaedalusGame commented:
Probably, but how? I'd say it'd be possible to add an electrolysis reaction of sorts to something like grounding rods when they get hit with tesla lightning.

Buggy123 commented:
>According to the defines, tritium is the most carbon you can get

What?
![](https://imgur.com/yAO8y1n.jpg)

W-what?

![](https://imgur.com/8xxYKwT.jpg)

What?!

![](https://imgur.com/4VK7bcQ.jpg)

# What.

arsserpentarium commented:
maybe when emitter rays penetrates gas?
also good idea to burn H+CO2-->H2O+welding_fuel_vapour

DaedalusGame commented:
Dunc said that gas types are expensive on performance so I'll stray from cluttering this pr with another gas, i already had to do minor changes to reactions to realize hydrogen detonation.

Emitter Rays/Particle Accelerator penetrating gas should be left for fusion or similar I think. Electrolysis by radiation sounds sketchy af

F-OS commented on code/modules/atmospherics/machinery/portable/canister.dm:
```diff
@@ -137,6 +137,12 @@
 
+/obj/machinery/portable_atmospherics/canister/hydrogen
+	name = "hydrogen canister"
+	desc = "Hydrogen. Who would win in a fight, a tetravalent main component of all life or one thicc proton??"
```
thicc

AverageJoe82 commented:
H+CO2-->H2O+welding_fuel_vapour should be 2H+CO2 -->H2O+CO(carbon monoxide) realistically, but I don't think it would naturally flow in that direction or if it would need activation energy

nicbn commented:
Hydrogen gas is H2 not H

DaedalusGame commented:
I love that we found so many people who know so much about a topic i know very little about: molecular organic chemistry

nicbn commented:
# HYDROGEN IS INORGANIC

DaedalusGame commented:
but hydrocarbons are organic

nicbn commented:
Hydro*carbons* are not hydrogen
Basically it is organic only when there is carbon.

DaedalusGame commented:
i'm convinced that welding fuel vapor would be a hydrocarbon

nicbn commented:
Yes but hydrogen wouldn't
Please do not add more reactions than the burn in this PR

nicbn reviewed (changes requested):
Make the reagent hydrogen spawn hydrogen gas when spilled (1 unit is 1 mol of gas)

DaedalusGame commented:
I'm adding electrolysis and nobody can stop me from making mad scientist setups involving tesla coils a reality

Buggy123 commented:
So, I was thinking about what burning hydrogen should look like, and then I realized that at any concentration where it could burn, a even mixture of hydrogen/oxygen will always explode. And then when I thought about it more, I realized this applies to pretty much _all_ flammable gasses and flames from bunsen burners, etc are only flames because the two gases aren't pre-mixed; all plasma fires should in reality be plasma _explosions_.

What I'm trying to say is I have no idea what hydrogen fires should look like because they're fictional, so I'm not going to be much help there, sorry.

DaedalusGame commented:
Aka make it look like normal fire gotcha boss

AverageJoe82 commented:
fuck im actually retarded, thanks @nicbn

nicbn commented:
Don't add electrolysis
Believe me, hydrogen is too op to be obtainable via reactions
It is odorless, colorless, nontoxic.

DaedalusGame committed:
- Forgot giving this new explosion gas to my favorite job at roundstart
- Makes hydrogen explosion produce water vapor too so you can't make infinite explosion rooms

DaedalusGame commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1777,7 +1789,7 @@
 	description = "A colorless liquid that suppresses violence on the subjects. Cheaper to synthetize, but wears out faster than normal Pax."
 	metabolization_rate = 1.5 * REAGENTS_METABOLISM
-  
+
```
ok i guess

DaedalusGame commented:
Not allowing an atmos gas to be produced by reaction is an affront to our gods

arsserpentarium commented:
@AverageJoe82,I know, but it is 2d space game.And welding fuel is already gas.

arsserpentarium commented:
@DaedalusGame If you add electrolysis, you should add tesla dischargers.

DaedalusGame commented:
You can use tesla coils, and welding fuel is not a gas

duncathan commented:
ughh i need to think about this one. still not even convinced i want it yet

arsserpentarium commented:
1.found it,
2.I forgot, it was on polaris.

DaedalusGame commented:
that's why it's DNM tagged, Dunc

MoreRobustThanYou commented:
@duncathan wait for proper SM interactions before judging please

DaedalusGame commented:
I was thinking something like that, also adding MORE radiation to tritium makes hydrogen which is less radioactive really makes you think

arsserpentarium commented:
not sure about second.But it could  be fairly interesting scheme, if we will use hydrogen for "homemade" rad collector.and process tritium to energy somehow.

DaedalusGame committed:
Merge branch 'master' of https://github.com/tgstation/tgstation into haidara

# Conflicts:
#	code/modules/atmospherics/gasmixtures/reactions.dm

optimumtact commented:
closing until dunc wakes up from his eternal sleep

Buggy123 commented:
He was arguing with me earlier today dunc wat
